package/MDAnalysis/analysis/msd.py

```diff
@@ -344,9 +344,14 @@ def _prepare(self):
         # these need to be zeroed prior to each run() call
         self.results.msds_by_particle = np.zeros((self.n_frames,
                                                   self.n_particles))
-        self._position_array = np.zeros(
-            (self.n_frames, self.n_particles, self.dim_fac))
+        # self._position_array = np.zeros(
+        #     (self.n_frames, self.n_particles, self.dim_fac))
         # self.results.timeseries not set here
+        start, stop, step = self._trajectory.check_slice_indices(self.start, self.stop, self.step)
+        self._position_array = self._trajectory.timeseries()
+        self._position_array = self._position_array[start: stop: step]
+        self._position_array = np.transpose(self._position_array, (1, 0, 2))
```

Comment:
You can use the `order` argument to `timeseries` to control the output order so that you don't need this transpose.

Reply:
Thank you for your suggestions! I will try to fix them.